### PR TITLE
Add Python Code for DFS Solution (620)

### PR DESCRIPTION
**Pull request by IceSerpent**
For the self-study video for this problem

_If any of the below doesn't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions), which includes the following: 
  - I understand that if it is clear that I have not attempted to follow these guidelines (ex. if I have not used tabs to indent), my PR will be closed.
  - If changes are requested, I will re-request the review after making them.

### CODE REVIEW

File: solutions/usaco-620.mdx [hunk 1]
```diff
@@ -107,6 +107,51 @@ public class pails {
 
 </JavaSection>
 
+<PySection>
+
+```py
+filein = open("pails.in", 'r')
+x, y, k, m = map(int, filein.read().split())
+
+sol = 1e18
```
Comment (SansPapyrus683) [inline solutions/usaco-620.mdx:116]: use `float('inf')` for effectively infinite values

File: solutions/usaco-620.mdx [hunk 2]
```diff
@@ -107,6 +107,51 @@ public class pails {
...
+
+vis = [[[False for a in range(101)] for b in range(101)] for c in range(101)]
```
Comment (SansPapyrus683) [inline solutions/usaco-620.mdx:118]: put 100 in a variable called `MAX_OPS` or something

File: solutions/usaco-620.mdx [hunk 3]
```diff
@@ -107,6 +107,51 @@ public class pails {
...
+
+def ff (curX, curY, curK):
```
Comment (SansPapyrus683) [inline solutions/usaco-620.mdx:120]: type annotations

File: solutions/usaco-620.mdx [hunk 4]
```diff
@@ -107,6 +107,51 @@ public class pails {
...
+    global sol
+
+    if curK > k or vis[curX][curY][curK]:
+        return
+    
+    vis[curX][curY][curK] = True
+    sol = min(sol, abs(m - (curX + curY)))
+
+    # case 1
```
Comment (SansPapyrus683) [inline solutions/usaco-620.mdx:129]: what are these "cases" you speak of?

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

File: solutions/usaco-620.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply solutions/usaco-620.mdx:129]: nvm

File: solutions/usaco-620.mdx [hunk 5]
```diff
@@ -107,6 +107,53 @@ public class pails {
...
+filein = open("pails.in", 'r')
+x, y, k, m = map(int, filein.read().split())
+MAX_OPS = 101
+
+sol = float('inf')
+
+vis = [[[False for a in range(MAX_OPS)] for b in range(MAX_OPS)] for c in range(MAX_OPS)]
+
+
+def ff(curX:'int', curY:'int', curK:'int'):
```
Comment (ryanchou-dev) [inline solutions/usaco-620.mdx:122]:
```suggestion
def ff(curX: 'int', curY: 'int', curK: 'int'):
```

File: solutions/usaco-620.mdx [hunk 6]
```diff
@@ -107,6 +107,53 @@ public class pails {
...
+        ff(0, curY + curX, curK + 1)
+
+    if curY > x - curX:
+        ff(x, curY - x + curX, curK + 1)
+    else:
+        ff (curX + curY, 0, curK + 1)
+
+
+ff (0, 0, 0)
+print(sol, file = open("pails.out", "w"))
```
Comment (ryanchou-dev) [inline solutions/usaco-620.mdx:152]:
```suggestion
        ff(curX + curY, 0, curK + 1)


ff(0, 0, 0)
print(sol, file=open("pails.out", "w"))
```

File: solutions/usaco-620.mdx [hunk 7]
```diff
@@ -107,6 +107,52 @@ public class pails {
 
 </JavaSection>
 
+<PySection>
+
+```py
+MAX_OPS = 101
```
Comment (SansPapyrus683) [inline solutions/usaco-620.mdx:113]: change to 100 and use `MAX_OPS + 1`

File: solutions/usaco-620.mdx [hunk 8]
```diff
@@ -107,6 +107,52 @@ public class pails {
...
+
+x, y, k, m = map(int, open('pails.in', 'r').read().split())
+
+sol = float('inf')
+vis = [[[False for a in range(MAX_OPS)] for b in range(MAX_OPS)] for c in range(MAX_OPS)]
```
Comment (SansPapyrus683) [inline solutions/usaco-620.mdx:118]: over 80 chars

File: solutions/usaco-620.mdx [hunk 9]
```diff
@@ -107,6 +107,52 @@ public class pails {
...
+
+
+def ff(curX: int, curY: int, curK: int) -> int:
```
Comment (SansPapyrus683) [inline solutions/usaco-620.mdx:121]: use snake_case

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by SansPapyrus683 — APPROVED** [review]

**Review by ryanchou-dev — APPROVED** [review]
⭐

**Pull request merged**